cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -109,4 +114,26 @@ public void getData(HttpRequest request, HttpResponder responder,
     ApplicationId application = namespace.app(previewId);
     responder.sendString(HttpResponseStatus.OK, GSON.toJson(previewManager.getRunner(application).getData(tracerId)));
   }
+
+  @POST
+  @Path("/previews/{preview-id}/tracers/")
+  public void getTracersData(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @PathParam("preview-id") String previewId) throws Exception {
+    NamespaceId namespace = new NamespaceId(namespaceId);
+    ApplicationId application = namespace.app(previewId);
+    Map<String, List<String>> previewRequest = parseBody(request,
+                                                         new TypeToken<Map<String, List<String>>>() { }.getType());
+    if (previewRequest == null) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Tracer name is not provided.");
+      return;
+    }
+    Map<String, Map<String, List<JsonElement>>> result = new HashMap<>();
+    for (List<String> tracerNames : previewRequest.values()) {
```

Comment:
We actually do not care about the keys. We only care about the list of tracer names that are passed in. So the body can be `{
  "tracers": ["tracer1", "tracer2", ...] 
}` or it can be `{
  "input": ["tracer1", "tracer2", ...], "output": ["tracer3", "tracer4"]
}`